### DISCUSSION TITLE
Blazor and Maui

### DISCUSSION THREAD

**Discussion by jmscraig** [MicroTrendsLtd/NinjaTrader8#58] [General]
I have rooting around on these topics and don't have a lot of time tonight but wanted to reply a little ... in a separate issue to be closed just so I don't clutter the thread with your last post.

1) I found and joined your discord server as Hedge.  building up a nice server there.

2) Looks like you guys primarily focus on the ES market.   I have not had time and wont have time soon to sort out ES myself.. too many opportunities.  If I find something I can really use without distracting me from my primary focus areas I do not mind being one of your DIY type funded customers. 

3) You mentioned a an interest in Maui as a primary UI.  All the Microsoft mentions I have seen on Maui highlight use on an iPhone.  Do you feel Maui is a great go-forward UI for Microsoft Server 2019, Microsoft Server 2022, and Windows 10  & 11 clients?

4) Your new alpha website looks great.   Last night I walked through the site, the Help page, reviewed the mouse over definitions and a number of historic website pages and pdfs but still did not find definitions for all the terms.  

In this image what does U720, M5 and P24 stand for?    What words would you use to describe your term "Blackhole?"
![UMP-Balckhole](https://user-images.githubusercontent.com/15313578/153976914-18d1cae1-9443-4dcf-8a96-52cbb63a33c3.jpg)

5) I saw your alpha site uses Blazor.  I thought you might have interest in this project that integrates into Blazor multiple real-time websock price subscription feeds and takes advantage of Microsoft's RX offering.   The "Order Book L3" portion of this website was never built out. 
https://crypto.mkotas.cz/
https://github.com/Marfusios/crypto-watcher-blazor

**Comment by MicroTrendsTom** [comment]
1) Nice welcome the idea is to provide free info and tools and a self-assisted model with community collaboration etc. – there are help sections and links off to help desks also – with free daily trade coaching- the DIY model I got that idea from Azure in fact – and sometimes I do subscribe to monthly support with Azure but mostly its DIY roll up sleeves and research etc.
2)  Micro Futures Indices are the focus for the trading group the poll section shows why, RTY, NQ, YM, ES, MNQ is far the most popular, volumes and range show etc. US indices are the most popular and liquid day trading instruments and easiest to make consistent money on due to range, news cycles and well established market behavior's, patterns and price cycles - and the NYSE session open.
3)  100% MAUI is the replacement for WPF on all device’s windows, Linux, app desktop - and mobile android, MS, and apple. But it is a bit too bleeding edge for me to fully engage in it yet and had several complications last year on release dates. But saying that with .net 6 and VS2022 out its about time to start – but I wanted to wait for the first service pack for VS2022 or make sure its tenable etc.
   The MS dev net webinars really show the power on this -   As for Apple phone yes you can also focus on platform specifics within sections for  that target platform if you want  to - other than that it is generic finally 1 code base for all devices and OS, awesome - yes I’m totally in 1000%  and this is the future for the next dare I say 25 to 50 years.
.NET Core MAUI revitalized my interest in development as did Blazor shame we did not have it all in about 2009 when WPF was hatching out mainstream. I was so fed up with the Hodge podge of mess in development until recently I wanted to quit – then along came net core 5 and MAUI – YES!

4. Awesome – yes, it’s always good to work server side and some client-side stuff can be fun. Blazor is well matured by now -, .net 6 really hits it out the park out the box it has all I need - but then I can add in a hub for SignalR should I need more WebSocket’s and simply send a message... and use an Azure service to allow massive user connections np. or I can wheel out Azure Service Bus and so on and of course auto scaling features and load balancing cloud based its incredibly easy – just love it.

Within the trading group there are sections that explain the “AWT” information and examples of trading with it. the essence is – are we mixed messy or uniform one color to identify a choppy or a correlated purposeful directional market. You can hit me up in the room, so I know your username and help orientate.

For the order book stuff, I bet that is good, but I do not look at 3rd party apps -especially where an overlap and conflict of what is on my to do list which stretches into a 3 year to 5 sprint – Order Flow is 10000% on that. But my own flavor – 80% of my codebase was taken offline in fact in 2016 to give me a breather from tech support – so I got a shed load of stuff waiting to come out again for this new gen of products – that can be community self-supported etc.
 But conceptually yes 10000% I am totally interested in the mechanics of that link but cannot look at the implementation, as conceptually and visually the result might be a conflicting overlap, the design presentation but the code behind nuts and bolts I would of course love to indulge a look see but will refrain at this point.
. Sounds pretty funky and even I say so with Alphwebtrader  it has some pretty funky stuff going on with WebSocket’s, webapis, service bus, SignalR, auto scaling load balancing and distributed caching so It can scale up to 100K connections as a target right now, but it could do a lot more of course but I doubt it will need more than 1000 capacity time will tell,  a mobile (MAUI app) is due out before 2023 the target 100K downloads   so all I do is to deformalize a lot of data and send it in chunks “views” to clients  therefore the logic is  server side and clients can be r2apidly made just to consume a chuck of data from an API ,WebSocket hub or message q etc.	   and in addition then desktop apps can be open source as the secret source is server side – tis just a questions of securing endpoints to prevent attacks and abuse, DDOS and so on.
